core/src/types.ts

```diff
@@ -299,16 +299,17 @@ export declare interface VaultTransactionFees {
     transactionFeeLiquidationDai: BigNumber;
 }
 
-export declare interface Vault extends VaultBase, VaultAmount, LiquidationLimits {}
+export declare interface Vault extends VaultBase, VaultAmount, VaultCollateralParameters, LiquidationLimits {}
 export declare interface OraclePrices {
-    currentUnitPrice: BigNumber;
-    nextUnitPrice: BigNumber;
-    nextPriceChange: Date;
+    // can be set to NaN
+    currentUnitPrice: BigNumber | number;
+    nextUnitPrice: BigNumber | number;
+    nextPriceChange?: Date;
```

Comment:
Date can also be invalid btw, `new Date(NaN)`, but we don't handle it automatically in the components yet